mParticle-Radar/MPKitRadar.m

```diff
@@ -103,28 +102,77 @@ - (MPKitExecStatus *)didBecomeActive {
     return [[MPKitExecStatus alloc] initWithSDKCode:[MPKitRadar kitCode] returnCode:MPKitReturnCodeSuccess];
 }
 
-#pragma mark Events
+#pragma mark User attributes and identities
+
+- (NSString*) getMpId:(FilteredMParticleUser *)user {
+    if (user != nil && user.userId != nil && user.userId.integerValue != 0) {
+        return [user.userId stringValue];
+    } else {
+        return nil;
+    }
+}
+
+- (void)setMparticleMetadata:(NSString *)mpId {
+    if ([Radar getMetadata] != NULL) {
+        NSMutableDictionary *userMetadata = [[Radar getMetadata] mutableCopy];
+        [userMetadata setObject:mpId forKey:@"mParticleId"];
+        NSDictionary *userMetadataDict = [userMetadata copy];
+        [Radar setMetadata:userMetadataDict];
+    }
+    else {
+        NSDictionary *userMetadataDict = @{@"mParticleId" : mpId};
+        [Radar setMetadata:userMetadataDict];
+    }
+}
 
-- (MPKitExecStatus *)logout {
+- (void)setRadarCustomerId:(FilteredMParticleUser *)user {
+    NSString *customerId = [user.userIdentities objectForKey:[NSNumber numberWithInt:MPUserIdentityCustomerId]];
+    if (customerId) {
+        [Radar setUserId:customerId];
+    }
+}
+
+- (MPKitExecStatus *)onLoginComplete:(FilteredMParticleUser *)user request:(FilteredMPIdentityApiRequest *)request {
+    NSString *mpId = [self getMpId:user];
+    [self setMparticleMetadata:mpId];
+    [self setRadarCustomerId:user];
     if (runAutomatically) {
-        [Radar stopTracking];
+        [self tryTrackOnce];
+        [self tryStartTracking];
     }
-    
     return [[MPKitExecStatus alloc] initWithSDKCode:[MPKitRadar kitCode] returnCode:MPKitReturnCodeSuccess];
 }
 
-#pragma mark User attributes and identities
+- (MPKitExecStatus *)onLogoutComplete:(FilteredMParticleUser *)user request:(FilteredMPIdentityApiRequest *)request {
+    NSString *mpId = [self getMpId:user];
+    [self setMparticleMetadata:mpId];
+    [self setRadarCustomerId:user];
+    if (runAutomatically) {
+        [self tryTrackOnce];
+        [self tryStartTracking];
+    }
+    return [[MPKitExecStatus alloc] initWithSDKCode:[MPKitRadar kitCode] returnCode:MPKitReturnCodeSuccess];
+}
 
-- (MPKitExecStatus *)setUserIdentity:(NSString *)identityString identityType:(MPUserIdentity)identityType {
-    if (identityType == MPUserIdentityCustomerId) {
-        [Radar setUserId:identityString];
-        
-        if (runAutomatically) {
-            [self tryTrackOnce];
-            [self tryStartTracking];
-        }
+- (MPKitExecStatus *)onIdentifyComplete:(FilteredMParticleUser *)user request:(FilteredMPIdentityApiRequest *)request {
+    NSString *mpId = [self getMpId:user];
+    [self setMparticleMetadata:mpId];
+    [self setRadarCustomerId:user];
```

Comment:
If we're always setting `userId` and `metadata` together, should we just combine these methods into a single `setRadarUser` function put the logic in one place and reduce margin for error? Or `setUserAndTrack` like on Android? Can we name everything similarly across both kits to make the logic easier to follow?